### ISSUE TITLE
Segfault in tst/testinstall/syntaxtree.tst

### ISSUE THREAD

**Issue by alex-konovalov** [gap-system/gap#3187]
This happens in the master branch when all packages are loaded. See e.g. on Travis here: https://travis-ci.org/gap-infra/gap-docker-master-testsuite/jobs/480427212. Started to happen two days ago.

**Comment by alex-konovalov** [comment]
Incidentally, on the same day Semigroups and Utils tests started to fail too:
- https://travis-ci.org/gap-infra/gap-docker-pkg-tests-master/jobs/480676994
- https://travis-ci.org/gap-infra/gap-docker-pkg-tests-master/jobs/480677009
for the same versions of these packages for which they used to work a day before.

**Comment by alex-konovalov** [comment]
The error happening in Semigroups does not sound right at all:
```
######## > Diff in:
/home/gap/inst/gap-master/pkg/semigroups-3.0.20/tst/standard/congrms.tst:104
# Input is:
IsSubrelation(cong2, cong);
# Expected output:
Error, Semigroups: IsSubrelation: usage,
congruences must be defined over the same semigroup,
# But found:
Error, Z(3)^7Z(3)^7
########
#I  Errors detected while testing
```

**Comment by ChrisJefferson** [comment]
EDIT: No, that's old code.

**Comment by ChrisJefferson** [comment]
Updated problem (I think)

`GET_VALUE_FROM_CURRENT_BODY` isn't wired up properly to the syntaxtree (I think), so getting constants isn't producing the right answers.

**Comment by fingolfin** [comment]
`GET_VALUE_FROM_CURRENT_BODY` is fine, the problem is that I introduced a silly GC bug, by keeping a pointer into an object across an allocation. Should be fixed in PR #3191
